pkg/util/hlc/hlc.go

```diff
@@ -30,6 +30,24 @@ import (
 	"github.com/pkg/errors"
 )
 
+const (
+	// The following constants are used if server.hlc.persist_future_wall_time
+	// is true
+
+	// persistFutureWallTimeInterval is the interval used for persisting the
+	// future wall time
+	persistFutureWallTimeInterval = 400 * time.Millisecond
+	// persistFutureWallTimeDelta is the delta used to compute the future wall
+	// time. This should be greater than persistFutureWallTimeInterval
+	persistFutureWallTimeDelta = int64(15 * time.Second)
```

Comment:
These constants will need to be tuned. Making them configurable through a cluster setting makes things a bit ugly.

A high `persistFutureWallTimeDelta` also delays CRDB restarts (as it will wait till future wall time) before proceeding.

A low `persistFutureWallTimeDelta` can cause spurious crashes if CPU usage is high and the persist gets delayed.